### PR TITLE
ARROW-16588: [C++][FlightRPC] Don't subclass GTest in test helpers

### PR DESCRIPTION
**Pull request by lidavidm**
Also, don't link every Arrow library to UCX when enabled.

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -827,11 +827,6 @@ if(ARROW_USE_XSIMD)
   list(APPEND ARROW_STATIC_LINK_LIBS xsimd)
 endif()
 
-if(ARROW_WITH_UCX)
-  list(APPEND ARROW_LINK_LIBS ucx::ucx)
-  list(APPEND ARROW_STATIC_LINK_LIBS ucx::ucx)
-endif()
-
```
Comment (cyb70289) [inline cpp/CMakeLists.txt:834]:
```
In file included from ../src/arrow/flight/transport/ucx/flight_transport_ucx_test.cc:33:
../src/arrow/flight/transport/ucx/ucx_internal.h:27:10: fatal error: 'ucp/api/ucp.h' file not found
#include <ucp/api/ucp.h>
         ^~~~~~~~~~~~~~~
1 error generated.
```

Comment (kou) [reply cpp/CMakeLists.txt:834]: Could you try the following patch?

```diff
diff --git a/cpp/src/arrow/flight/transport/ucx/CMakeLists.txt b/cpp/src/arrow/flight/transport/ucx/CMakeLists.txt
index 71392ec0af..d682ead633 100644
--- a/cpp/src/arrow/flight/transport/ucx/CMakeLists.txt
+++ b/cpp/src/arrow/flight/transport/ucx/CMakeLists.txt
@@ -53,6 +53,7 @@ if(ARROW_BUILD_TESTS)
         arrow_flight_static
         arrow_flight_testing_static
         arrow_flight_transport_ucx_static
+        ucx::ucx
         ${ARROW_TEST_LINK_LIBS})
   else()
     set(ARROW_FLIGHT_UCX_TEST_LINK_LIBS
@@ -60,6 +61,7 @@ if(ARROW_BUILD_TESTS)
         arrow_flight_shared
         arrow_flight_testing_shared
         arrow_flight_transport_ucx_shared
+        ucx::ucx
         ${ARROW_TEST_LINK_LIBS})
   endif()
   add_arrow_test(flight_transport_ucx_test
```

Comment (cyb70289) [reply cpp/CMakeLists.txt:834]: It works. Thanks @kou!

**Review by kou — APPROVED** [review]
+1

Thanks!

**Pull request closed without merging**